### PR TITLE
Replace string comparison with FileSystem utils

### PR DESCRIPTION
**Pull request by TheOneRing**
Needs more work,,,

### CODE REVIEW

File: src/common/utility.cpp [hunk 1]
```diff
@@ -277,15 +279,18 @@ bool Utility::fsCasePreserving()
...
 bool Utility::fileNamesEqual(const QString &fn1, const QString &fn2)
 {
-    const QDir fd1(fn1);
-    const QDir fd2(fn2);
-
+    const QFileInfo fd1(fn1);
+    const QFileInfo fd2(fn2);
+    qDebug() << fd1.canonicalFilePath() << fd2.canonicalFilePath();
     // Attention: If the path does not exist, canonicalPath returns ""
     // ONLY use this function with existing pathes.
```
Comment (fmoc) [inline src/common/utility.cpp:286]: I think the comment should be reworded a bit, now that this is actually checked (which generally is a good idea; comments in the implementation aren't really read by users).

File: src/common/utility.cpp [hunk 2]
```diff
@@ -277,15 +279,18 @@ bool Utility::fsCasePreserving()
...
+    Q_ASSERT(fd1.exists());
+    Q_ASSERT(fd2.exists());
+    Q_ASSERT(fd1.canonicalFilePath() == fn1);
```
Comment (fmoc) [inline src/common/utility.cpp:289]: You are calculating these canonical paths below anyway.

Comment (TheOneRing) [reply src/common/utility.cpp:289]: ITS A WIP
and the idea is that I can remove the cannonical path one day

Comment (TheOneRing) [reply src/common/utility.cpp:289]: I just created it to fix finish it one day

**Pull request closed without merging**